### PR TITLE
Add Player Status & more Buttons

### PR DESCRIPTION
**Pull request by Botond24**
I updated this plugin with multiple part to extend the experience:
- is_playing variable
- is_shuffle variable
- playback_status variable
- loop_status variable
- Skip Forward Action
- Skip Backward Action
- Loop Switching Actions
- Shuffle On/Off Actions
- updated Play Pause Action
- updated Next Action
- updated Prev Action

### CODE REVIEW

File: Actions/MediaFastForwardAction.cs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    public MediaFastForwardAction()
+    {
+        if (PluginManager.Plugins.ContainsKey("Macro Deck Media Controls"))
+        {
+            MacroDeckPlugin plugin = PluginManager.Plugins["Macro Deck Media Controls"];
+            if (plugin.GetType().IsEquivalentTo(typeof(MediaControlsPlugin)))
+            {
+                _manager = ((MediaControlsPlugin)plugin).Manager;
+            }
+        }
```
Comment (manuelmayer-dev) [inline Actions/MediaFastForwardAction.cs]: You don't need to make this that complicated, you can make a public static instance of the plugin in the MediaControlsPlugin class and then access it.

Example: https://github.com/manuelmayer-dev/Macro-Deck-Windows-Utils-Plugin/blob/master/Main.cs

File: Actions/MediaFastForwardAction.cs [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    public override string Name => "Media Fast Forward";
+    public override string Description => "Fast forwards the current track on a media player.\n\r\n\rConfiguration: no";
+    public override void Trigger(string clientId, ActionButton actionButton)
+    {
+        var session = _manager.GetCurrentSession();
+        if (session == null)
+        {
+            return;
+        }
+        var test = Task.Run(async () => await session.TryFastForwardAsync()).GetAwaiter().GetResult();
```
Comment (manuelmayer-dev) [inline Actions/MediaFastForwardAction.cs]: You can remove the variable and the .GetAwaiter().GetResult(); is also not necessary.

File: Actions/MediaLoopList.cs [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    public MediaLoopList()
+    {
+        if (PluginManager.Plugins.ContainsKey("Macro Deck Media Controls"))
+        {
+            MacroDeckPlugin plugin = PluginManager.Plugins["Macro Deck Media Controls"];
+            if (plugin.GetType().IsEquivalentTo(typeof(MediaControlsPlugin)))
+            {
+                _manager = ((MediaControlsPlugin)plugin).Manager;
+            }
+        }
```
Comment (manuelmayer-dev) [inline Actions/MediaLoopList.cs]: Same here and in the other classes you created.

File: Actions/MediaLoopList.cs [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    public override string Name => "Media Loop List";
+    public override string Description => "Turns on looping for the current playlist on a media player.\n\r\n\rConfiguration: no";
+    public override void Trigger(string clientId, ActionButton actionButton)
+    {
+        var session = _manager.GetCurrentSession();
+        if (session == null)
+        {
+            return;
+        }
+        var test = Task.Run(async () => await session.TryChangeAutoRepeatModeAsync(MediaPlaybackAutoRepeatMode.List)).GetAwaiter().GetResult();
```
Comment (manuelmayer-dev) [inline Actions/MediaLoopList.cs]: Same here and in the other classes you created.

**Review by manuelmayer-dev — CHANGES_REQUESTED** [review]

File: Actions/MediaFastForwardAction.cs (hunk 2 above)
Comment (Botond24) [reply Actions/MediaFastForwardAction.cs]: Thank you for the help

**Comment by manuelmayer-dev** [comment]
@Botond24 The pr has merge conflicts

**Pull request merged**